dask_match/io/parquet.py

```diff
@@ -62,6 +71,23 @@ class ReadParquet(IO):
     def engine(self):
         return get_engine("pyarrow")
 
+    @property
+    def input_columns(self):
+        return self.operands[self._parameters.index("columns")]
+
+    @property
+    def columns(self):
+        if self.input_columns is None:
+            return self._meta.columns
+        else:
+            import pandas as pd
+
+            return pd.Index(_list_columns(self.input_columns))
```

Comment:
Note that overlapping parameters and properties can be a bit confusing, and I don't really like the solution used here.

We may want to require each class to define a set of reserved names, and forbid any parameter name from intersecting with that set. This means the user-facing `read_parquet` API, for example, would need to translate arguments like `columns` into a *different* name (like `column_projection`), so that the `ReadParquet` implementation wouldn't need to worry about implementing anything like the `input_columns` workarounds used here.